### PR TITLE
Add error handling when procfile-parse returns empty string

### PR DESCRIPTION
**Pull request by karanthukral**
## What?
- `procfile-start` throws the error `setuidgid: usage: setuidgid account child` when its called with an unknown process type.
  - This is because `procfile-parse` returns an empty string leading to the `setuidgid` command failing in `procfile-exec`
  - The error is not clear and does not point to the issue being with the Procfile

## Solution
- This PR adds simple error checking to the `procfile-start` function that errors out if the provided process type is not defined in the Procfile. If the process type is found, the function behaves as before

### CODE REVIEW

**Comment by josegonzalez** [comment]
Can we have a test-case for this?

**Comment by karanthukral** [comment]
> Can we have a test-case for this?
Where do you think the test case for this should go? Should it be part if a particular buildpack test or within the root test file (https://github.com/gliderlabs/herokuish/blob/master/buildpacks/test)?

**Comment by joshmanders** [comment]
probably make it in the root test file, right? As it's not exclusive to a specific buildpack.

**Comment by josegonzalez** [comment]
I defer to @michaelshobbs here tbh.

**Comment by karanthukral** [comment]
I added a fairly simple test in `buildpacks/test` to basically run the `/start` command with an invalid procfile process and confirm the error message is as expected. Let me know if you have any other ideas for possible tests?

File: buildpacks/test [hunk 1]
```diff
@@ -20,9 +20,20 @@ run-test() {
   cd - &> /dev/null
 	[[ "$CI" ]] || rmflag="--rm"
   [[ "$TRACE" ]] && debug_flag="-e TRACE=true"
+  run-invalid-proc-test
   docker run $rmflag $debug_flag --env=USER=herokuishuser -v "$app_path:/tmp/app" herokuish:dev /bin/herokuish test / "$app"
 }
 
+run-invalid-proc-test() {
```
Comment (michaelshobbs) [inline buildpacks/test]: Can you please move the test execution to `tests/functional/tests.sh`?

**Comment by karanthukral** [comment]
@michaelshobbs I moved the tests as per your recommendation

File: tests/functional/tests.sh [hunk 2]
```diff
@@ -32,3 +32,12 @@ T_default-user() {
 	}
 	herokuish-test "test-user" "$(fn-source _test-user)"
 }
+
+T_invalid_proc_process() {
+  local expected_err_msg="Proc entrypoint invalid-proc does not exist. Please check your Procfile"
+  local err_msg=$(herokuish-test "inavlid-proc" "herokuish procfile start invalid-proc")
```
Comment (michaelshobbs) [inline tests/functional/tests.sh]: typo? "inavlid-proc"

Comment (karanthukral) [inline tests/functional/tests.sh]: I'll fix that and the test output

**Comment by karanthukral** [comment]
`cat: /app/Procfile: No such file of directory` was thrown from [here](https://github.com/gliderlabs/herokuish/blob/3921f4fe51999f49ae11d91008200a6204cdb5d8/include/procfile.bash#L24). The `procfile-parse` expects that file to exist but doesn't error without it. The test passed because the error check this PR puts into place is still emitted. 

In order to write this test with a `Procfile` present in the app path, would it be valid to have a test Procfile with in the functional tests directory which can be mounted onto the container?

@michaelshobbs

**Comment by michaelshobbs** [comment]
I'd probably just echo a Procfile out in the test and clean it up at the end.

**Comment by karanthukral** [comment]
@michaelshobbs Let me know if the test seems valid now.

File: tests/functional/tests.sh [hunk 3]
```diff
@@ -32,3 +42,19 @@ T_default-user() {
...
+	
+	echo "Creating Procfile"
+	echo "web:" > $dir/"Procfile"
+	
+	local expected_err_msg="Proc entrypoint invalid-proc does not exist. Please check your Procfile"
+  	local err_msg="$(docker run $rmflag $debug_flag --env=USER=herokuishuser -v "$dir:/tmp/app" herokuish:dev /start invalid-proc)"
+ 	
+	if [[ $err_msg != $expected_err_msg ]]; then
+		echo "procfile-start did not throw error for invalid procfile"
+    	exit 1
```
Comment (michaelshobbs) [inline tests/functional/tests.sh]: nitpick: indentation is off

Comment (karanthukral) [inline tests/functional/tests.sh]: Fixed 👍

**Comment by michaelshobbs** [comment]
Awesome thanks for the contribution!

**Pull request merged**